motionplan/ik/inverseKinematics.go

```diff
@@ -16,18 +19,79 @@ const (
 	defaultGoalThreshold = defaultEpsilon * defaultEpsilon
 )
 
-// InverseKinematics defines an interface which, provided with seed inputs and a Metric to minimize to zero, will output all found
+// InverseKinematics defines an interface which, provided with seed inputs and a function to minimize to zero, will output all found
 // solutions to the provided channel until cancelled or otherwise completes.
 type InverseKinematics interface {
-	referenceframe.Limited
```

Comment:
Not sure what the benefit of removing some of our types is, which seems to be a general theme of the changes in this package.  If we are writing this as a generic optimization package then its fine to have concepts like this one in it since it doesnt specifically relate to frames and just uses a concept that also applies to them.

Reply:
Hmm, that's fair.
I do want to move this towards a more generic optimization package, but you're right, limits are still relevant. I'll add back.

Reply:
Limits I agree stand on their own, Inputs do not.